test/SpokePoolClient.deposits.ts

```diff
@@ -1,83 +1,105 @@
-      await simpleDeposit(spokePool, erc20, depositor1, depositor1, destinationChainId2),
-      await simpleDeposit(spokePool, erc20, depositor2, depositor2, destinationChainId),
-      await simpleDeposit(spokePool, erc20, depositor2, depositor2, destinationChainId2),
-      await simpleDeposit(spokePool, erc20, depositor2, depositor2, destinationChainId2),
-    ];
-    await spokePoolClient.update();
+  it("Correctly fetches deposit data multiple fills, multiple chains", async function () {
+    const deposit: Deposit = {
+      depositId: 0,
+      depositor: depositor.address,
+      recipient: depositor.address,
+      originToken: erc20.address,
+      amount: toBNWei("1"),
+      originChainId,
+      destinationChainId,
+      relayerFeePct: toBNWei("0.01"),
+      quoteTimestamp: Date.now(),
+      realizedLpFeePct: toBNWei("0.01"),
+      destinationToken: destErc20.address,
+      message: "0x",
+    };
 
-    const outputDeposits = Object.fromEntries(
-      [destinationChainId, destinationChainId2].map((_destinationChainId) => {
-        const destinationChainId = Number(_destinationChainId);
-        const deposits = spokePoolClient.getDepositsForDestinationChain(destinationChainId);
-        expect(deposits.length).to.equal(3);
-        return [destinationChainId, deposits];
-      })
-    );
+    // Do 6 deposits. 2 for the first depositor on chain1, 1 for the first depositor on chain2, 1 for the second
+    // depositor on chain1, and 2 for the second depositor on chain2.
+    await buildFill(spokePool, destErc20, depositor, relayer1, deposit, 0.1);
+    await buildFill(spokePool, destErc20, depositor, relayer1, deposit, 0.1);
+    await buildFill(spokePool, destErc20, depositor, relayer1, { ...deposit, originChainId: originChainId2 }, 0.1);
+
+    await buildFill(spokePool, destErc20, depositor, relayer2, deposit, 0.1);
+    await buildFill(spokePool, destErc20, depositor, relayer2, { ...deposit, originChainId: originChainId2 }, 0.1);
+    await buildFill(spokePool, destErc20, depositor, relayer2, { ...deposit, originChainId: originChainId2 }, 0.1);
+
+    await spokePoolClient.update();
 
     // Validate associated ChainId Events are correctly returned.
-    expect(Object.values(outputDeposits).flat().length).to.equal(Object.keys(inputDeposits).length);
-    inputDeposits.forEach((inputDeposit) => {
-      const depositId = inputDeposit.depositId as number;
-      const destinationChainId = inputDeposit.destinationChainId as number;
-      const outputDeposit = outputDeposits[destinationChainId].find((deposit) => deposit.depositId === depositId);
-      Object.entries(inputDeposit).forEach(([k, v]) => {
-        expect(v).to.not.be.undefined;
-        expect(v).to.deep.equal(outputDeposit?.[k]);
-      });
-    });
+    expect(spokePoolClient.getFills().length).to.equal(6);
+
+    // TODO: Add `getFillsForRepaymentChainId` tests once we update the `fillRelay` method from contracts-v2 to allow
```

Comment:
Is this a TODO you added? 

Also, I think we should try to remove the dependency on any functions from `contracts-v2` so if that means copying the `fiillRelay` code (and all other code in that repo) into this repo, it'd be better. I've found it frustrating in the past to not be able to modify the code because its from another repo. Same reasoning as the @uma functions :)

Reply:
No that was also ported over. I think that may be better suited to do this in another PR.